### ISSUE TITLE
[no-conditional-expect] how to catch errors for testing

### ISSUE THREAD

**Issue by dankolesnikov** [jest-community/eslint-plugin-jest#866]
Hi everyone,

When I was implementing testing with Jest for Axios, I've noticed that this is regarded as anti pattern:
```
    try {
      await axios(url);
    } catch (error) {
      expect(error.response.status).toBe(404);
    }
```

Why is that? We need that because this is the only way to capture the entire `error` object. And assert on objects within.
Current functionality of Jest for capturing error only allows to see the message of the exception not the entire error object.

**Comment by G-Rath** [comment]
This is laid out in the docs for [`no-conditional-expect`](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-conditional-expect.md#rule-details). It's because Jest only considers a test to have failed if it throws an error.

For example with your code, if `await axios(url)` is successful, the test will be marked as passing despite having not called `expect`.
The quickest way to fix the underlying error would be to add a throw in the `try`:

```
try {
  await axios(url);
  throw new Error('unexpected success');
} catch(error) {
  expect(error.response.status).toBe(404);
}
```

However that is still considered a conditional by our rule, which wants you to get the `expect` out of the `catch` all together - I outlined a good way of doing this with a wrapper [here](https://github.com/jest-community/eslint-plugin-jest/issues/295#issuecomment-583788671), which I've distilled into [a PR](https://github.com/jest-community/eslint-plugin-jest/pull/867) adding this to the docs for `no-conditional-expect` :)
